### ISSUE TITLE
A problem about scripts on macOS 10.13 High Sierra with CotEditor 3.2.3

### ISSUE THREAD

**Issue by tatsy-ymj** [coteditor/CotEditor#739]
I still have the problem with scripts.

1. Open a markdown text file, and run the following ruby script. It works as I intend.

```
#!/usr/bin/env ruby -Ku
#%%%{CotEditorXInput=AllText}%%%
#%%%{CotEditorXOutput=Pasteboard}%%%

while $stdin.gets
print "- " + $1 if /^## ?([^#]+)/ =~ $_
end
```

2. The above script works well with text files which have less than 21962 characters (UTF-8), but CotEditor freezes with text files which have more than 21963 characters. 

I attach test files and a movie.
[script_test.zip](https://github.com/coteditor/CotEditor/files/1406593/script_test.zip)
https://drive.google.com/file/d/0B0emCo_bF-igSGYzSWwxaXVpclk/view?usp=sharing

**Comment by tatsy-ymj** [comment]
cf. #733

**Comment by 1024jp** [comment]
Thank you for the report and sorry for another bug.
I believe I could fix the issue this time.
The fix will be shipped with CotEditor 3.2.4, which will be released in early November.
